### PR TITLE
Add notes for action.openPopup API in Firefox

### PR DESCRIPTION
**Pull request by oliverdunk**
#### Summary

Updates the compatibility information for the WebExtension `action.openPopup` API following the changes in https://phabricator.services.mozilla.com/D139796. In particular, Firefox now has support for a `windowId` parameter and the user gesture requirement has been removed behind a preference currently enabled in Nightly.

I have a similar PR coming to update the main MDN page, which this one is intended to land at the same time as.

#### Test results and supporting details

Original bug (including `windowId` support): https://bugzilla.mozilla.org/show_bug.cgi?id=1755763

Removing preference: https://bugzilla.mozilla.org/show_bug.cgi?id=1799344

I've been working on these changes with @Rob--W.

#### Related issues

NA

### CODE REVIEW

File: webextensions/api/action.json [hunk 1]
```diff
@@ -238,7 +238,11 @@
               },
               "edge": "mirror",
               "firefox": {
-                "version_added": "101"
+                "version_added": "101",
+                "notes": [
+                  "Must be called from inside the handler for a <a href='https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/User_actions'>user action</a>. From Firefox 108, this restriction has been lifted behind a preference. See <a href='https://bugzil.la/1799344'>bug 1799344</a>.",
```
Comment (Rob--W) [inline webextensions/api/action.json:243]: Ideally, I'd just shorten this to "Since Firefox 108, user interaction is no longer required to call this API."

Now, since the pref is required, maybe also `... when the ... pref is set (<a ...>bug xxx</a>).`

When the pref is finally flipped, we could do something like this:
https://github.com/mdn/browser-compat-data/blob/2687c37c6450a1c4c323370cb9bb1a486416fbd4/webextensions/api/identity.json#L63-L67

... or just keep the minimal simple text from the top of this comment, without any reference to a preference, since the preference (flag) would be irrelevant: https://github.com/mdn/browser-compat-data/blob/main/docs/data-guidelines/index.md#removal-of-irrelevant-flag-data

**Comment by queengooborg** [comment]
Hey @oliverdunk, I just noticed that this was your first PR for BCD -- welcome to the project (sorry that you weren't welcomed sooner)!  If you have any questions, don't hesitate to ask a project peer or owner!

**Comment by oliverdunk** [comment]
Thanks @queengooborg! And apologies for the delay, I’m slowly catching up on responding to feedback 😅

File: webextensions/api/action.json (hunk 1 above)
Comment (oliverdunk) [reply webextensions/api/action.json]: Thanks for the feedback, I pushed a change :)

**Review by Rob--W — APPROVED** [review]

**Review by queengooborg — APPROVED** [review]
Thank you for the quick changes, let's get this merged!
